domain/tutoringModel/ga.py

```diff
@@ -0,0 +1,312 @@
+            self.evolve(fitness, best_sample)
+
+        ga_path = np.array(self.learning_elements)
+        population = self.valide_population()
+        idx = population[0]
+        result_ga_lp = ga_path[idx]
+
+        # convert to string
+        learning_path = self.get_learning_path_as_str(result_ga_lp)
+
+        return learning_path
+
+    def get_learning_path_as_str(self, result_ga):
+        """.."""
+        str_learning_path = ""
+        contain_le = False
+        for ele in result_ga:
+            str_learning_path = str_learning_path + ele + ", "
+            contain_le = True
+
+        if contain_le:
+            str_learning_path = str_learning_path[:-2]
+
+        return str_learning_path
+
+    def get_learning_style(self, learning_style):
+        """..."""
+
+        new_learning_style = {}
+        str_processing = learning_style.get("processing_dimension")
+        value_processing = learning_style.get("processing_value")
+        new_learning_style[str_processing] = value_processing
+
+        str_perception = learning_style.get("perception_dimension")
+        value_perception = learning_style.get("perception_value")
+        new_learning_style[str_perception] = value_perception
+
+        str_input = learning_style.get("input_dimension")
+        value_input = learning_style.get("input_value")
+        new_learning_style[str_input] = value_input
+
+        str_understanding = learning_style.get("understanding_dimension")
+        value_understanding = learning_style.get("understanding_value")
+        new_learning_style[str_understanding] = value_understanding
+
+        for key, value in dict(new_learning_style).items():
+            if value is None:
+                del new_learning_style[key]
+
+        return new_learning_style
+
+    def check_learning_style(self, input_learning_style):
+        """check_learning_style"""
+        is_correct = False
+        for iterator in input_learning_style:
+            if input_learning_style.get(iterator):
+                dimension_number = input_learning_style.get(iterator)
+                if dimension_number < 0 or dimension_number > 11:
+                    is_correct = True
+                    break
+
+        return is_correct
+
+    def check_name_learning_style(self, input_learning_style):
+        """this ist no necesary"""
+
+        list_is_correct = []
+        for iterator in input_learning_style:
+            condition1 = iterator == "act" or iterator == "ref"
+            condition2 = iterator == "sns" or iterator == "int"
+            condition3 = iterator == "vis" or iterator == "vrb"
+            condition4 = iterator == "seq" or iterator == "glo"
+            if condition1 or condition2 or condition3 or condition4:
+                list_is_correct.append(True)
+        temp = [True, True, True, True]
+        if list_is_correct != temp:
+            return True
+
+        return False
+
+    def get_learning_element(self, learning_elements):
```

Comment:
add function description

Reply:
I agree and have added a comment

Reply:
Ive noticted that the function is named "get_learning_element" but returns a list. A more concise naming would be "convert_learning_element_dic_to_list" or something similar. Currently it conveys the message that it returns a single element.